exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/exporter.rb

```diff
@@ -0,0 +1,303 @@
+          private_constant(:ERROR_MESSAGE_INVALID_HEADERS)
+
+          def self.ssl_verify_mode
+            if ENV.key?('OTEL_RUBY_EXPORTER_OTLP_SSL_VERIFY_PEER')
+              OpenSSL::SSL::VERIFY_PEER
+            elsif ENV.key?('OTEL_RUBY_EXPORTER_OTLP_SSL_VERIFY_NONE')
+              OpenSSL::SSL::VERIFY_NONE
+            else
+              OpenSSL::SSL::VERIFY_PEER
+            end
+          end
+
+          def initialize(endpoint: config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_EXPORTER_OTLP_ENDPOINT', default: 'http://localhost:4318/v1/traces'), # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+                         certificate_file: config_opt('OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE', 'OTEL_EXPORTER_OTLP_CERTIFICATE'),
+                         ssl_verify_mode: Exporter.ssl_verify_mode,
+                         headers: config_opt('OTEL_EXPORTER_OTLP_TRACES_HEADERS', 'OTEL_EXPORTER_OTLP_HEADERS', default: {}),
+                         compression: config_opt('OTEL_EXPORTER_OTLP_TRACES_COMPRESSION', 'OTEL_EXPORTER_OTLP_COMPRESSION', default: 'gzip'),
+                         timeout: config_opt('OTEL_EXPORTER_OTLP_TRACES_TIMEOUT', 'OTEL_EXPORTER_OTLP_TIMEOUT', default: 10),
+                         metrics_reporter: nil)
+            raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}" if invalid_url?(endpoint)
+            raise ArgumentError, "unsupported compression key #{compression}" unless compression.nil? || %w[gzip none].include?(compression)
+
+            @uri = if endpoint == ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
+                     URI("#{endpoint}/v1/traces")
+                   else
+                     URI(endpoint)
+                   end
+
+            @http = http_connection(@uri, ssl_verify_mode, certificate_file)
+
+            @path = @uri.path
+            @headers = case headers
+                       when String then parse_headers(headers)
+                       when Hash then headers
+                       else
+                         raise ArgumentError, ERROR_MESSAGE_INVALID_HEADERS
+                       end
+            @timeout = timeout.to_f
+            @compression = compression
+            @metrics_reporter = metrics_reporter || OpenTelemetry::SDK::Trace::Export::MetricsReporter
+            @shutdown = false
+          end
+
+          # Called to export sampled {OpenTelemetry::SDK::Trace::SpanData} structs.
+          #
+          # @param [Enumerable<OpenTelemetry::SDK::Trace::SpanData>] span_data the
+          #   list of recorded {OpenTelemetry::SDK::Trace::SpanData} structs to be
+          #   exported.
+          # @param [optional Numeric] timeout An optional timeout in seconds.
+          # @return [Integer] the result of the export.
+          def export(span_data, timeout: nil)
+            return FAILURE if @shutdown
+
+            send_bytes(OpenTelemetry::Exporter::OTLP::Common.encode(span_data), timeout: timeout)
```

Comment:
Do you prefer this to including the `Common` module in this class?

Reply:
I do prefer this over including and having the implicit availability of an `encode` method.

Reply:
Dummy question: which one is more performant? D'you know if this is encoded as 4 constant lookups followed by a send?

Reply:
Conclusion is that it doesn't matter, both because this isn't a hot path and because the chain of constant lookups is cached.